src/node/node_info_network.h

```diff
@@ -18,7 +18,115 @@ namespace ccf
     std::string rpcport;
     std::string pubport;
   };
-  DECLARE_JSON_TYPE(NodeInfoNetwork);
+  DECLARE_JSON_TYPE(NodeInfoNetwork_v1);
   DECLARE_JSON_REQUIRED_FIELDS(
-    NodeInfoNetwork, rpchost, pubhost, nodehost, nodeport, rpcport, pubport);
+    NodeInfoNetwork_v1, rpchost, pubhost, nodehost, nodeport, rpcport, pubport);
+
+  struct NodeInfoNetwork_v2
+  {
+    struct NetAddress
+    {
+      std::string hostname;
+      std::string port;
+
+      bool operator==(const NetAddress& other) const
```

Comment:
Can be default now.

Reply:
I think if we start using `= default` we're enforcing C++20 (rather than supporting both C++20 and C++17), which we don't want to do yet?

Reply:
True, although we could also ifdef and it'd make the old stuff easier to drop later.